Composer/packages/client/src/components/CreationFlow/v2/CreateOptions.tsx

```diff
@@ -0,0 +1,252 @@
+        },
+      },
+    },
+  };
+};
+
+const tableCell = css`
+  outline: none;
+  :focus {
+    outline: rgb(102, 102, 102) solid 1px;
+  }
+`;
+
+const content = css`
+  outline: none;
+`;
+
+const optionKeys = {
+  createFromScratch: 'createFromScratch',
+  createFromQnA: 'createFromQnA',
+  createFromTemplate: 'createFromTemplate',
+};
+
+// -------------------- CreateOptions -------------------- //
+type CreateOptionsProps = {
+  templates: BotTemplate[];
+  onDismiss: () => void;
+  onNext: (data: string) => void;
+  fetchTemplates: (feedUrls?: string[]) => Promise<void>;
+  fetchReadMe: (moduleName: string) => Promise<void>;
+  templateReadMe: string;
+} & RouteComponentProps<{}>;
+
+export function CreateOptionsV2(props: CreateOptionsProps) {
+  const [option] = useState(optionKeys.createFromTemplate);
+  const [disabled] = useState(false);
+  const { templates, onDismiss, onNext, fetchReadMe, templateReadMe } = props;
+  const [currentTemplate, setCurrentTemplate] = useState('');
+  const [emptyBotKey, setEmptyBotKey] = useState('');
+  const [selectedFeed, setSelectedFeed] = useState<{ props: IPivotItemProps } | undefined>(undefined);
+
+  const selection = useMemo(() => {
+    return new Selection({
+      onSelectionChanged: () => {
+        const t = selection.getSelection()[0] as BotTemplate;
+        if (t) {
+          setCurrentTemplate(t.id);
+        }
+      },
+    });
+  }, []);
+
+  const handleJumpToNext = () => {
+    let routeToTemplate = emptyBotKey;
+    if (option === optionKeys.createFromTemplate) {
+      routeToTemplate = currentTemplate;
+    }
+
+    if (option === optionKeys.createFromQnA) {
+      routeToTemplate = QnABotTemplateId;
+    }
+
+    if (props.location && props.location.search) {
+      routeToTemplate += props.location.search;
+    }
+
+    onNext(routeToTemplate);
+  };
+
+  const tableColums = [
+    {
+      key: 'name',
+      name: formatMessage('Name'),
+      fieldName: 'name',
+      minWidth: 150,
+      maxWidth: 200,
+      isResizable: !disabled,
+      data: 'string',
+      styles: rowTitle(disabled),
+      onRender: (item) => (
+        <div data-is-focusable css={tableCell}>
+          <div css={content} tabIndex={-1}>
+            {item.name}
+          </div>
+        </div>
+      ),
+    },
+  ];
+
+  const onRenderRow = (props) => {
+    if (props) {
+      return (
+        <DetailsRow {...props} data-testid={props.item.id} styles={rowDetails(disabled)} tabIndex={props.itemIndex} />
+      );
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    if (templates.length > 1) {
+      const emptyBotTemplate = find(templates, ['id', EmptyBotTemplateId]);
+      if (emptyBotTemplate) {
+        setCurrentTemplate(emptyBotTemplate.id);
+        setEmptyBotKey(emptyBotTemplate.id);
+      }
+    }
+  }, [templates]);
+
+  useEffect(() => {
+    if (selectedFeed?.props?.headerText?.toLowerCase() === 'typescript') {
```

Comment:
There must be a better way to access the value of the selected feed.

Also, we should pull these feeds out into a dictionary rather than define them inline here.  

`props.fetchTemplates(feeds[selectedFeed])`

Reply:
maybe at least use the `itemKey` field instead of the actual label text.

Eventually these feeds will be user modifiable, so they will have a key, title, and url value.